src/SDKs/Azure.Core/data-plane/Azure.Core/Pipeline/Policies/BearerTokenAuthenticationPolicy.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Azure.Core.Credentials;
+using Azure.Core.Pipeline;
+
+namespace Azure.Base.Http.Pipeline
+{
+    public class BearerTokenAuthenticationPolicy : HttpPipelinePolicy
+    {   
+        private TokenCredential _credential;
+        private string[] _scopes;
+
+        public BearerTokenAuthenticationPolicy(TokenCredential credential, string scope)
+        {
+            _credential = credential ?? throw new ArgumentNullException(nameof(credential));
+            _scopes = new string[] { scope } ?? throw new ArgumentNullException(nameof(scope));
+        }
+
+        public override async Task ProcessAsync(HttpPipelineMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
+        {
+            string token = await _credential.GetTokenAsync(_scopes, message.Cancellation).ConfigureAwait(false);
+            
+            message.Request.AddHeader(HttpHeader.Names.Authorization, "Bearer " + token);
```

Comment:
Should we add tests, asserts, or even pipeline feature to ensure that some policies are never after retry? We don't want the same header to be added several times.

@pakrym?

Reply:
> We don't want the same header to be added several times

We should use SetHeader then. Can't the token might change in between retries?